### PR TITLE
feat(upgrade): merge upgrade-job crate to upgrade

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
bors try

File: k8s/upgrade/src/upgrade_resources/mod.rs
Comment (tiagolobocastro) [inline k8s/upgrade/src/upgrade_resources/mod.rs:1]: This file can be deleted now?

Comment (sinhaashish) [reply k8s/upgrade/src/upgrade_resources/mod.rs:1]: yes, because `upgrade_resources` is not a folder any more. all the code has been removed to `upgrade_plugin`

**Review by niladrih — COMMENTED** [review]
Why not put the upgrade-job in the /bin directory (ref: https://doc.rust-lang.org/cargo/guide/project-layout.html)?

Also, this change looks like it is moving the upgrade-job into the upgrade directory. Why is this necessary? Should this instead be a refactor, so that the upgrade lib contains all of the shared resources which can be used by upgrade-job and upgrade-plugin? Isn't this the same without this change? What am I missing here?

**Review by niladrih — COMMENTED** [review]
Why not put the upgrade-job in the /bin directory (ref: https://doc.rust-lang.org/cargo/guide/project-layout.html)?

Also, this change looks like it is moving the upgrade-job into the upgrade directory. Why is this necessary? Should this instead be a refactor, so that the upgrade lib contains all of the shared resources which can be used by upgrade-job and upgrade-plugin? Isn't this the same without this change? What am I missing here?

**Comment by sinhaashish** [comment]
> Why not put the upgrade-job in the /bin directory (ref: https://doc.rust-lang.org/cargo/guide/project-layout.html)?
> 
> Also, this change looks like it is moving the upgrade-job into the upgrade directory. Why is this necessary? Should this instead be a refactor, so that the upgrade lib contains all of the shared resources which can be used by upgrade-job and upgrade-plugin? Isn't this the same without this change? What am I missing here?

I had tried that in the previous version of my PR , wherein the errors and commons were in a common folder and were used by both upgrade job and plugin. That didn't worked , cargo clippy had errors stating that there were some unused code when i compiled the binary i.e. upgrade-job, and if the library were compiled then again there were errors of unused code from binary. 
Thats the reason, upon discussion with @tiagolobocastro  i just merged the two folders into one.
Upgrade functionality under one folder is an ideal situation which this PR is doing. Upgrade job and plugin both inside the upgrade folder.

File: k8s/upgrade/src/plugin/preflight_validations.rs [hunk 1]
```diff
@@ -216,3 +216,26 @@ pub async fn upgrade_path_validation(namespace: &str) -> error::Result<()> {
...
+        &self.client
+    }
+}
+
+impl RestClient {
+    /// Create new Rest Client from the given `Configuration`.
+    pub fn new_with_config(config: Configuration) -> RestClient {
+        Self {
+            client: tower::ApiClient::new(config),
+        }
```
Comment (niladrih) [inline k8s/upgrade/src/plugin/preflight_validations.rs:239]: Why is this change required?

**Review by niladrih — COMMENTED** [review]
Left one comment.

File: k8s/upgrade/src/plugin/preflight_validations.rs (hunk 1 above)
Comment (niladrih) [reply k8s/upgrade/src/plugin/preflight_validations.rs:239]: Oh, okay

**Review by niladrih — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors merge
